_posts/2019-10-29-dhis2-and-docker.md

```diff
@@ -0,0 +1,214 @@
+layout: post
+categories: [blog, docker]
+author: gintare
+---
+
+# DHIS2 with docker
+Docker makes core-teams development and testing efforts easier every day! It allows us to spin up various environments with any DHIS2 version, numerous tomcat versions and different components, such as redis, nginx, in no time. Because we wanted to share docker with community, we made publishing docker images a part of our delivery pipeline. This guide aims to provide some guidelines on how to use docker to quickly set up DHIS2. 
+
+## Prerequisites
+This guide is aimed at people with at least minimal knowledge about docker. If you don't know too much about docker just yet, but think that it might be useful to you, I encourage you to take a look at resources listed in [resources](#Resources) section. 
+
+In addition to minimal docker-knowledge, you will need: 
+- docker downloaded and installed on your machine. 
+- docker-compose downloaded and installed on your machine. 
+
+## Introduction
+
+### Vocabulary
+*Image* - a template that contains the application and all the dependencies that are required to run that application. 
+*Container* - a running instance of docker image. 
+*Dockerfile* - text file, with lines and instructions on how to build the docker image. 
+
+### Docker Hub
+
+Docker Hub is an official cloud-based repository where docker users store and distribute their images. DHIS2 has its organization on [Docker Hub](https://hub.docker.com/u/dhis2/). At the time of writing this guide, we have 4  **active** repositories: base-dev, base, core-dev, core.
+
+#### Repositories
+Sometimes we categorize repositories in _channels_ and say that we have 2 image channels: _dev_ and _stable_. 
+##### dev
+Repositories suffixed with `-dev` contains images for _latest_ DHIS2 versions. That means, that we can't fully guarantee its quality and we recommend to use it only for testing purposes. Also, for curious souls, wanting to peek into the latest DHIS2 development efforts. The `-dev` repositories are ideologically the same as `dev` demo instances on _play_. Images in this channel will be overwritten multiple times a day. 
+
+##### stable
+Repositories **not** suffixed by `dev` contains images for **stable** and **released** DHIS2 versions. These images represent the stable dhis2 versions, meaning it won't t be rebuilt in the future. 
+
+##### core
+Core repositories (core, core-dev) contains tomcat images that already have DHIS2 war baked-in.  That means, that starting this image will result in a running DHIS2 instance. 
+
+##### base
+Base repositories (base, base-dev) contains only DHIS2 war. Starting this image will not result in a running DHIS2 instance. 
+
+#### Versioning and tags
+Docker tags are meant to convey useful information about a specific version/variant of the image. In DHIS2 repositories, we combine DHIS2 and tomcat versions with image variants into a tag to be as transparent as possible about what you will find in the image.
```

Comment:
The descriptions here are really good to have, and in addition we can link to how we name WAR and Docker images as a reference: https://github.com/dhis2/notes/blob/master/decisions/2019/05/25-war-docker-schemes.md